### PR TITLE
Allow multiple master selection

### PR DESCRIPTION
**Pull request by kiall**
Allow the user to choose multiple masters during the initial
bootstrap.

We don't currently allow for adding a master post bootstrap.

Depends-On: https://github.com/kubic-project/automation/pull/40
Depends-On: https://github.com/kubic-project/automation/pull/60

### CODE REVIEW

**Comment by kiall** [comment]
The UX here really isn't great, I'd like to see someone more familiar with HTML/JS etc update the UI to change over to use a dropdown / select on the right hand side to choose between "Unused" / "Master" / "Worker", and the column of checkboxes on the left be removed entirely.

File: app/assets/javascripts/dashboard.js [hunk 1]
```diff
@@ -538,26 +557,29 @@ $('body').on('change', 'input[name="roles[worker][]"]', function() {
...
-    var $checkbox = $(this).closest('tr').find('input[name="roles[worker][]"]');
+    MinionPoller.selectedMasters.push(value);
 
     $checkbox.prop('checked', true);
     $checkbox.prop('disabled', true);
+  } else {
+    var index = MinionPoller.selectedMasters.indexOf(value);
+    MinionPoller.selectedMasters.splice(index, 1);
+
+    $checkbox.prop('disabled', false);
```
Comment (vitoravelino) [inline app/assets/javascripts/dashboard.js]: We need to maintain the previous behavior of unchecking the worker/node checkbox if it wasn't previously checked before selecting the node as master. Renaming the variable to `$workerCheckbox` would also help the code understanding.

Scenario 1:

1. No nodes selected yet
1. Select a node as master
1. Unselect as master the node from previous step 
1. Worker checkbox should not be selected

Scenario 2:

1. A node is selected to bootstrap
1. Select a node as master
1. Unselect as master the node from previous step 
1. Worker checkbox should be kept as checked

We can write a feature test for this if it doesn't exist yet for further refactoring.

**Comment by vitoravelino** [comment]
> The UX here really isn't great, I'd like to see someone more familiar with HTML/JS etc update the UI to change over to use a dropdown / select on the right hand side to choose between "Unused" / "Master" / "Worker", and the column of checkboxes on the left be removed entirely.

Sounds like a good idea. I can help you with this. @mssola, what do you think?

If we go with this approach, you can ignore the comment I made previously.

**Comment by vitoravelino** [comment]
> The UX here really isn't great, I'd like to see someone more familiar with HTML/JS etc update the UI to change over to use a dropdown / select on the right hand side to choose between "Unused" / "Master" / "Worker", and the column of checkboxes on the left be removed entirely.

Just realized that I saw a [tip](https://uxdesign.cc/design-better-forms-96fadca0f49c#1a5f) a while ago and we can go further and get rid of the dropdown and show the 3 options as button groups. What do you think?

**Comment by kiall** [comment]
> Just realized that I saw a tip a while ago and we can go further and get rid of the dropdown and show the 3 options as button groups. What do you think?

I think that makes sense, assuming your talking about the "Show all selection options if under 6" tip.

**Comment by mssola** [comment]
@vitoravelino @kiall this change on the design makes sense and it shouldn't break too much what we had before 👍

**Comment by kiall** [comment]
Good State: 3+3 - worked
image

Good State: 3+1 - worked
image

Bad State 3+0 worked:
image

Bad State 2+3 worked:

image

Good State 1+3 worked:

**Comment by kiall** [comment]
Bad State 1+1 worked:

image

Good State 1+2 worked:

image

**Comment by kiall** [comment]
Validated locally, works nicely!

**Pull request merged**